cognite/client/_api/datapoints_subscriptions.py

```diff
@@ -0,0 +1,259 @@
+
+        Args:
+            external_id (str | Sequence[str]): External ID or list of external IDs of subscriptions to retrieve.
+            ignore_unknown_ids (bool): Whether to ignore IDs and external IDs that are not found rather than throw an exception.
+
+        Returns:
+            The requested subscriptions.
+
+        Examples:
+
+        Retrieve a subscription by external ID:
+
+            >>> from cognite.client import CogniteClient
+            >>> c = CogniteClient()
+            >>> batch = c.time_series.subscriptions.retrieve("my_subscription")
+
+        """
+        if self.show_experimental_warning:
+            warn(self._warning_message, FutureWarning)
+
+        return self._retrieve_multiple(
+            list_cls=DataPointSubscriptionList,
+            resource_cls=DatapointSubscription,
+            identifiers=IdentifierSequence.load(external_ids=external_id),
+            ignore_unknown_ids=ignore_unknown_ids,
+        )
+
+    def update(self, update: DataPointSubscriptionUpdate) -> DatapointSubscription:
+        """`Update a subscriptions <https://pr-2221.specs.preview.cogniteapp.com/20230101-beta.json.html#tag/Data-point-subscriptions/operation/updateSubscriptions>`_
+
+        Update a subscription. Note that Fields that are not included in the request are not changed.
+        Furthermore, the subscription partition cannot be changed.
+
+        Args:
+            update: The subscription update.
+
+        Returns:
+            Updated subscription.
+
+        Examples:
+
+        Change the name of a preexisting subscription:
+
+            >>> from cognite.client import CogniteClient
+            >>> from cognite.client.data_classes import DataPointSubscriptionUpdate
+            >>> c = CogniteClient()
+            >>> update = DataPointSubscriptionUpdate("my_subscription").name.set("My New Name")
+            >>> updated = c.time_series.subscriptions.update(update)
+
+
+        Add a time series to a preexisting subscription:
+
+            >>> from cognite.client import CogniteClient
+            >>> from cognite.client.data_classes import DataPointSubscriptionUpdate
+            >>> c = CogniteClient()
+            >>> update = DataPointSubscriptionUpdate("my_subscription").time_series_ids.add("MyNewTimeSeriesExternalId")
+            >>> updated = c.time_series.subscriptions.update(update)
+
+        """
+        if self.show_experimental_warning:
+            warn(self._warning_message, FutureWarning)
+
+        return self._update_multiple(
+            items=update,
+            list_cls=DataPointSubscriptionList,
+            resource_cls=DatapointSubscription,
+            update_cls=DataPointSubscriptionUpdate,
+        )
+
+    def list_data(
+        self,
+        external_id: str,
+        partitions: Sequence[tuple[int, str] | int | DataPointSubscriptionPartition],
```

Comment:
Homogeneous collections are easier to reason about. Is there really a need for mixing them?
```suggestion
        partitions: Sequence[tuple[int, str]] | Sequence[int] | Sequence[DataPointSubscriptionPartition],
```